src/Build/BackEnd/Shared/BuildRequestConfiguration.cs

```diff
@@ -485,6 +485,18 @@ public void InitializeProject(BuildParameters buildParameters, Func<ProjectInsta
             ErrorUtilities.VerifyThrow(IsLoaded, $"This {nameof(BuildRequestConfiguration)} must be loaded at the end of this method");
         }
 
+        internal void CreateUniqueGlobalProperty()
+        {
+            // create a copy so the mutation does not leak into the ProjectInstance
+            _globalProperties = new PropertyDictionary<ProjectPropertyInstance>(_globalProperties);
+
+            // if you change the key also change the following so that it does not leak into string caches:
+            // Microsoft.Build.OpportunisticIntern.BucketedPrioritizedStringList.TryIntern
+            // Microsoft.Build.Construction.ProjectStringCache.Add
+            var key = $"ProjectInstance{Guid.NewGuid():N}";
```

Comment:
Should this name be prefixed with `MSBuild` like other reserved/magic properties?